contracts/TokenHandler.sol

```diff
@@ -32,7 +32,7 @@ contract TokenHandler is ITokenHandler, ITokenManagerType, ReentrancyGuard, Crea
      * @return address the address of the token.
      */
     // slither-disable-next-line locked-ether
-    function giveToken(bytes32 tokenId, address to, uint256 amount) external payable returns (uint256, address) {
+    function giveToken(bytes32 tokenId, address to, uint256 amount) external returns (uint256, address) {
```

Comment:
can you make sure out tests cover a call to `interchainTransfer` with non-zero gasValue so this gets triggered?